backend-db/Rhyolite/Backend/DB/LargeObjects.hs

```diff
@@ -149,8 +149,7 @@ instance (Monad m, PostgresLargeObject m) => PostgresLargeObject (ReaderT r m) w
     lift $ withLargeObject oid mode (\lofd -> runReaderT (f lofd) s)
 
 withStreamedLargeObject
-  :: (MonadIO m)
-  => LargeObjectId
+  :: LargeObjectId
   -> (LBS.ByteString -> IO ())
-  -> DbPersist Postgresql m ()
-withStreamedLargeObject oid f = liftWithConn $ \conn -> LO.withLargeObjectLBS conn (toOid oid) f
+  -> Serializable ()
+withStreamedLargeObject oid f = unsafeLiftDbPersist $ liftWithConn $ \conn -> LO.withLargeObjectLBS conn (toOid oid) f
```

Comment:
How does this behave in the context of a serialization error?

Reply:
Sigh, this is tricky. I'm tempted to use repeatable read for this, but perhaps `SERIALIZABLE READ ONLY DEFERRED` would be best if the transaction is truly read-only.

Reply:
Ok, this function `withStreamedLargeObject` now works with `DbPersist` directly like it used to and it's up to the caller to decide how the transaction is handled. I also added another method to `RunDb` which makes calling this function easy (types line up) but doesn't have retry/etc.